aws/src/main/java/org/apache/iceberg/aws/s3/BaseS3File.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.aws.s3;
+
+import software.amazon.awssdk.http.HttpStatusCode;
+import software.amazon.awssdk.services.s3.S3Client;
+import software.amazon.awssdk.services.s3.model.HeadObjectRequest;
+import software.amazon.awssdk.services.s3.model.HeadObjectResponse;
+import software.amazon.awssdk.services.s3.model.S3Exception;
+
+public abstract class BaseS3File {
+  private final S3Client client;
+  private final S3URI location;
+  private HeadObjectResponse metadata;
+
+  public BaseS3File(S3Client client, S3URI location) {
+    this.client = client;
+    this.location = location;
+  }
+
+  public String location() {
+    return location.toString();
+  }
+
+  public S3Client getClient() {
+    return client;
+  }
+
+  public S3URI getLocation() {
```

Comment:
This class already has `location()` above. How about a package-private `uri` method?

Reply:
updated